### PR TITLE
Add Zeitwerk loader support

### PR DESCRIPTION
**Pull request by nadjaheitmann**

### CODE REVIEW

**Review by dosas — CHANGES_REQUESTED** [review]
I do not get why the tests are failing due to backwards compatibility since the failing tests are all for ruby >=3.

Dropping the tests is not an option.

File: .rubocop.yml [hunk 1]
```diff
@@ -15,6 +15,7 @@ Metrics/BlockLength:
   Max: 50
   Exclude:
     - 'test/**/*.rb'
+    - 'lib/foreman_snapshot_management/engine.rb'
```
[the remark this answers is not in the document]
Comment (nadjaheitmann) [reply .rubocop.yml:18]: I think the metric is mainly to keep complexity for functions as simple as possible. As this not really complies with this rule, I think it is okay to add this file.

[the remark this answers is not in the document]
Comment (dosas) [reply .rubocop.yml:18]: I think in that case we should exclude the rule not the complete file.

**Comment by dosas** [comment]
> > I do not get why the tests are failing due to backwards compatibility since the failing tests are all for ruby >=3.
> 
> It is because I adjusted to work foreman_snapshot_management master branch to work Foreman develop which uses Zeitwerk. But Zeitwerk is not used for Foreman 3.11 and 3.12 . Hence, this is a breaking change.
> 
> The pipeline always checks against the master branch of this foreman_snapshot_management, no matter what happens in upstream Foreman. So the problem is that the pipeline somehow expect that the plugin is always backwards compatible which it is not. And what is okay, I think, the same happens for all upstream plugins.

In that case everything is working as expected (develop: green, 3.11 and 3.12: red) 
I would opt for feature branches until 3.13 is released.

**Comment by nadjaheitmann** [comment]
> In that case everything is working as expected (develop: green, 3.11 and 3.12: red) 

Definitely, we just need to decide how to deal with that.

> I would opt for feature branches until 3.13 is released.

I would suggest to bump a version before merging this and then add a feature branch `3.0-stable` .

**Comment by nadjaheitmann** [comment]
> > In that case everything is working as expected (develop: green, 3.11 and 3.12: red)
> 
> Definitely, we just need to decide how to deal with that.
> 
> > I would opt for feature branches until 3.13 is released.
> 
> I would suggest to bump a version before merging this and then add a feature branch `3.0-stable` .

Actually, there are no significant new features, so we probably do not need to bump a new version. I added a new branch for foreman_snapshot_management version 3.x.x which does not run tests on the develop branch https://github.com/ATIX-AG/foreman_snapshot_management/tree/3.0-stable.

**Review by dosas — APPROVED** [review]

**Pull request merged**